### PR TITLE
feat(markdown-nav): expose button click events

### PR DESCRIPTION
**Pull request by christianmemije**
## Description

- Exposes button click events
- Adds a test

#### Test Steps

- View events demo on http://localhost:4200/#/components/markdown-navigator/examples

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

### CODE REVIEW

File: src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-events/markdown-navigator-demo-events.component.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          markdownString: `[Go to Mars](#data={"planet": "mars"})`,
+        },
+        {
+          title: 'Jupiter',
+          markdownString: `[Go to Jupiter](#data={"planet": "Jupiter"})`,
+        },
+      ],
+    });
+    dialogRef.componentInstance.buttonClicked.subscribe((data: ITdFlavoredMarkdownButtonClickEvent) =>
+      this._snackBar.open(`Button clicked: ${JSON.stringify(data)}`),
```
Comment (kriswinbush) [inline src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-events/markdown-navigator-demo-events.component.ts]: need to add a duration to snackbar.

File: src/platform/markdown-navigator/markdown-navigator-window/markdown-navigator-window.component.ts [hunk 2]
```diff
@@ -36,6 +37,7 @@ export class TdMarkdownNavigatorWindowComponent {
 
   @Output() closed: EventEmitter<void> = new EventEmitter();
   @Output() dockToggled: EventEmitter<boolean> = new EventEmitter();
+  @Output() buttonClicked: EventEmitter<ITdFlavoredMarkdownButtonClickEvent> = new EventEmitter();
```
Comment (kriswinbush) [inline src/platform/markdown-navigator/markdown-navigator-window/markdown-navigator-window.component.ts]: Should update API README.md with new output info

File: src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-events/markdown-navigator-demo-events.component.ts (hunk 1 above)
Comment (christianmemije) [reply src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-events/markdown-navigator-demo-events.component.ts]: Done.

File: src/platform/markdown-navigator/markdown-navigator-window/markdown-navigator-window.component.ts (hunk 2 above)
Comment (christianmemije) [reply src/platform/markdown-navigator/markdown-navigator-window/markdown-navigator-window.component.ts]: Done.

**Pull request merged**